### DISCUSSION TITLE
generation usage should accept a list of dict for async generation

### DISCUSSION THREAD

**Discussion by dove-young** [scalar/scalar#761] [Ideas]
So far `usage` in `generation` is a dictionary.  There is a problem with it. 

Some LLM provide `async` feature in their `generation` function. It accept a list of prompts, and process them in a batch. Then it gives a list of generated responses. 

In this case there are a list of prompts has been processed, and a list of generation and token usage count are returned. And there is a single generate/completion call. So there would be a single `generation` object in the `trace`. 

But now it is not possible to record these list of usage data into a `usage` record in the `generation` object

**Comment by marclave** [comment]
Hey @dove-young are you having any issues with our ai content writer or needing any assistance with the api references :)

**Comment by marcklingen** [comment]
Sorry for the confusion @marclave, this idea is regarding https://github.com/langfuse/langfuse

I loved how you routed q&a and ideas to Discussions ([config.yml](https://github.com/scalar/scalar/blob/main/.github/ISSUE_TEMPLATE/config.yml)) and was a bit too quick when adding it to our repo without replacing the links accordingly. This was fixed today ([PR](https://github.com/langfuse/langfuse/pull/867)).

Scalar seems to be great, impressed by the UX and DX. We will move the Langfuse api ref from redoc to scalar once there is support for models in responses ([idea](https://github.com/scalar/scalar/discussions/645)).

@dove-young can you move this idea to the langfuse repo?

**Comment by dove-young** [comment]
sorry for bothering you. I have no idea about how this was routed here. I might clicked new issue button and Feature request link in langfuse repo.  This is for langfuse. sorry.
